napari/_vispy/app/backends/_qt.py

```diff
@@ -0,0 +1,765 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+"""
+Base code for the Qt backends. Note that this is *not* (anymore) a
+backend by itself! One has to explicitly use either PySide, PyQt4 or
+PyQt5. Note that the automatic backend selection prefers a GUI toolkit
+that is already imported.
+
+The _pyside, _pyqt4 and _pyqt5 modules will import * from this module,
+and also keep a ref to the module object. Note that if two of the
+backends are used, this module is actually reloaded. This is a sorts
+of poor mans "subclassing" to get a working version for both backends
+using the same code.
+
+Note that it is strongly discouraged to use the PySide/PyQt4/PyQt5
+backends simultaneously. It is known to cause unpredictable behavior
+and segfaults.
+"""
+
+from __future__ import division
+
+from time import sleep, time
+import os
+import sys
+import atexit
+import ctypes
+
+from vispy.util import logger
+from ..base import BaseApplicationBackend, BaseCanvasBackend, BaseTimerBackend
+from vispy.util import keys
+from vispy.ext.six import text_type
+from vispy.ext.six import string_types
+from vispy import config
+from . import qt_lib
+
+USE_EGL = config['gl_backend'].lower().startswith('es')
+
+# Get platform
+IS_LINUX = IS_OSX = IS_WIN = IS_RPI = False
+if sys.platform.startswith('linux'):
+    if os.uname()[4].startswith('arm'):
+        IS_RPI = True
+    else:
+        IS_LINUX = True
+elif sys.platform.startswith('darwin'):
+    IS_OSX = True
+elif sys.platform.startswith('win'):
+    IS_WIN = True
+
+# -------------------------------------------------------------------- init ---
+
+
+def _check_imports(lib):
+    # Make sure no conflicting libraries have been imported.
+    libs = ['PyQt4', 'PyQt5', 'PySide', 'qtpy']
+    libs.remove(lib)
+    for lib2 in libs:
+        lib2 += '.QtCore'
+        if lib2 in sys.modules:
+            raise RuntimeError(
+                "Refusing to import %s because %s is already "
+                "imported." % (lib, lib2)
+            )
+
+
+# Get what qt lib to try. This tells us wheter this module is imported
+# via _pyside or _pyqt4 or _pyqt5
+QGLWidget = object
+if qt_lib == 'pyqt4':
+    _check_imports('PyQt4')
+    if not USE_EGL:
+        from PyQt4.QtOpenGL import QGLWidget, QGLFormat
+    from PyQt4 import QtGui, QtCore, QtTest
+
+    QWidget, QApplication = QtGui.QWidget, QtGui.QApplication  # Compat
+elif qt_lib == 'pyqt5':
+    _check_imports('PyQt5')
+    if not USE_EGL:
+        from PyQt5.QtOpenGL import QGLWidget, QGLFormat
+    from PyQt5 import QtGui, QtCore, QtWidgets, QtTest
+
+    QWidget, QApplication = QtWidgets.QWidget, QtWidgets.QApplication  # Compat
+elif qt_lib == 'qtpy':
```

Comment:
from what I can tell, the only real changes to vispy are here and line 57, along with the additional qtpy backend file, correct?

Reply:
yes that's correct, it is a pretty minor change, but we just needed all the other code so that these files would get read